data-node/src/main/java/org/graylog/datanode/opensearch/OpensearchProcessImpl.java

```diff
@@ -214,6 +228,41 @@ private void configure() {
         );
     }
 
+    @VisibleForTesting
+    void checkConfiguredHeap() {
+        Size heap = Size.parse(configuration.getOpensearchHeap());
+        long heapBytes = heap.toBytes();
+        long freeMemory = getFreeMemory();
+        long memoryRatio = freeMemory / heapBytes;
+        if (memoryRatio > MEMORY_RATIO_THRESHOLD) {
+            LOG.warn("There appears to be about {} times more available memory than the heap size configured for this data node.", memoryRatio);
+            clusterEventBus.post(new DataNodeNotficationEvent(nodeId.getNodeId(), Notification.Type.DATA_NODE_HEAP_WARNING,
+                    Map.of("hostname", configuration.getHostname(), "memoryRatio", memoryRatio)));
+        }
+    }
+
+    /**
+     * Tries to determine the free memory available on the machine at the given time minus a buffer of 2 GB.
+     *
+     * @return free memory minus buffer of 2 GB
+     */
+    @SuppressForbidden("Deliberate use of com.sun package, handling exception if not available")
+    long getFreeMemory() {
+        try {
+            OperatingSystemMXBean osBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+            long buffer = 2 * 1024 * 1024 * 1024L;
+            return osBean.getFreeMemorySize() - buffer;
```

Comment:
This computation leads on my machine to negative value.  It doesn't break the logic, but feels wrong to operate with negative free memory values. Should we return zero if the free memory is lower than buffer?

![image](https://github.com/user-attachments/assets/fca336bc-5fd7-4ef6-9ee1-3eaa5549533d)

Reply:
I switched to using oshi for determining the memory. Can you recheck if this works?

Reply:
It works now for me and delivers reasonable, positive numbers.

Is the usage of non floating-point arithmetic a deliberate decision? 

```
long memoryRatio = freeMemory / heapBytes;
```

![image](https://github.com/user-attachments/assets/10a6eae2-3c1f-482a-aba9-da6823191817)

Reply:
Good catch. God, I hate math in Java 😉